src/rules/unnecessaryConstructorRule.ts

```diff
@@ -39,12 +57,49 @@ export class Rule extends Lint.Rules.AbstractRule {
     public static FAILURE_STRING = "Remove unnecessary empty constructor.";
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithFunction(sourceFile, walk);
+        const options: Options = {
+            unnecessarySuperCall:
+                this.ruleArguments.length !== 0 &&
+                (this.ruleArguments[0] as { "unnecessary-super-call"?: boolean })[
+                    "unnecessary-super-call"
+                ] === true,
+        };
+
+        return this.applyWithFunction(sourceFile, walk, options);
     }
 }
 
-const isEmptyConstructor = (node: ts.ConstructorDeclaration): boolean =>
-    node.body !== undefined && node.body.statements.length === 0;
+const containsStatements = (statements: ts.NodeArray<ts.Statement>) => statements.length > 0;
+
+const containsSuper = (statements: ts.NodeArray<ts.Statement>): boolean => {
+    for (const statement of statements) {
```

Comment:
Here, you loop over all statements, but `containsSuper` is only called if `statements` is length 1. You can just have this take in a single `statement: ts.Statement`.

Reply:
Done